substrate/frame/revive/src/lib.rs

```diff
@@ -528,6 +538,63 @@ pub mod pallet {
 	#[pallet::storage]
 	pub(crate) type OriginalAccount<T: Config> = StorageMap<_, Identity, H160, AccountId32>;
 
+	/// The events emitted by this pallet while executing the current inflight transaction.
+	///
+	/// The events are needed to reconstruct the ReceiptInfo, as they represent the
+	/// logs emitted by the contract. The events are consumed when the transaction is
+	/// completed and moved to the `InflightTransactions` storage object.
+	#[pallet::storage]
+	#[pallet::unbounded]
+	pub(crate) type InflightEvents<T: Config> =
+		CountedStorageMap<_, Identity, u32, Event<T>, OptionQuery>;
+
+	/// True if the events must be collected.
+	#[pallet::storage]
+	pub(crate) type CollectInflightEvents<T: Config> = StorageValue<_, bool, ValueQuery>;
+
+	/// The EVM submitted transactions that are inflight for the current block.
+	///
+	/// The transactions are needed to construct the ETH block.
+	///
+	/// This contains the information about transaction payload, transaction index within the block,
+	/// the events emitted by the transaction, the status of the transaction (success or not), and
+	/// the gas consumed.
+	#[pallet::storage]
+	#[pallet::unbounded]
+	pub(crate) type InflightTransactions<T: Config> = CountedStorageMap<
+		_,
+		Identity,
+		u32,
+		(Vec<u8>, u32, Vec<Event<T>>, bool, Weight),
+		OptionQuery,
+	>;
+
+	/// The current Ethereum block that is stored in the `on_finalize` method.
+	///
+	/// # Note
+	///
+	/// This could be further optimized into the future to store only the minimum
+	/// information needed to reconstruct the Ethereum block at the RPC level.
+	///
+	/// Since the block is convenient to have around, and the extra details are capped
+	/// by a few hashes and the vector of transaction hashes, we store the block here.
+	#[pallet::storage]
+	#[pallet::unbounded]
+	pub(crate) type EthereumBlock<T> = StorageValue<_, EthBlock, ValueQuery>;
+
+	/// Mapping for block number and hashes.
+	///
+	/// The maximum number of elements stored is capped by the block hash count `BLOCK_HASH_COUNT`.
+	#[pallet::storage]
+	pub type BlockHash<T: Config> = StorageMap<_, Twox64Concat, U256, H256, ValueQuery>;
+
+	/// The details needed to reconstruct the receipt info offchain.
+	///
+	/// This contains valuable information about the gas used by the transaction.
+	#[pallet::storage]
+	#[pallet::unbounded]
```

Comment:
In frame_system there is this comment let's add something similar here

```rust
	/// Events deposited for the current block.
	///
	/// NOTE: The item is unbound and should therefore never be read on chain.
	/// It could otherwise inflate the PoV size of a block.
```

also you need to kill this storage on_initialize